code/network/multi_ingame.cpp

```diff
@@ -1127,13 +1139,35 @@ void send_ingame_ships_packet(net_player *player)
 			continue;
 		}
 
+		SCP_vector<ubyte> ship_flags_out, obj_flags_out;
+
 		//  add the ship name and other information such as net signature, ship and object(?) flags.
 		p_type = INGAME_SHIP_NEXT;
 		ADD_DATA( p_type );
 		ADD_STRING( shipp->ship_name );
 		ADD_USHORT( Objects[so->objnum].net_signature );
-		ADD_ULONG( shipp->flags.to_u64() );
-		ADD_ULONG( Objects[so->objnum].flags.to_u64() );
+		
+		auto check = shipp->flags.to_ubyte_vector(ship_flags_out);
+		Assertion(check <= 255, "Somehow FSO thinks there are too many ship flags, %d * 8 to be exact. Please report!", static_cast<int>(check));
+
+		auto temp_size = static_cast<ubyte>(check);
+		ADD_DATA(temp_size);
+
+		for (auto& item : ship_flags_out){
+			ADD_DATA(item);
+		} 
+
+		check = shipp->flags.to_ubyte_vector(ship_flags_out);
+
+		Assertion(check <= 255, "Somehow FSO thinks there are too many object flags, %d * 8 to be exact. Please report!", static_cast<int>(check));
+
+		temp_size = static_cast<ubyte>(Objects[so->objnum].flags.to_ubyte_vector(obj_flags_out));
```

Comment:
This looks like a copy-paste error. Should it not look like above:
```suggestion
		check = Objects[so->objnum].flags.to_ubyte_vector(obj_flags_out);

		Assertion(check <= 255, "Somehow FSO thinks there are too many object flags, %d * 8 to be exact. Please report!", static_cast<int>(check));

		temp_size = static_cast<ubyte>(check);
```

Reply:
Yes, luckily this only affected error checking.